src/com/facebook/buck/jvm/java/intellij/IjModuleFactory.java

```diff
@@ -670,6 +683,7 @@ public void apply(TargetNode<JavaTestDescription.Arg> target, ModuleBuildContext
           true /* wantsPackagePrefix */,
           context);
       addCompiledShadowIfNeeded(target, context);
+      context.setLanguageLevel(getLevel(target.getConstructorArg().source));
```

Comment:
multiple TargetNodes in the same module could have different language levels. The way you've written it means "last one wins", at minimum keep them as a ImmutableSortedSet<String> and get the first/last one.